src/code/providers/lara-provider.ts

```diff
@@ -87,18 +124,18 @@ class LaraProvider extends ProviderInterface {
     }
   }
 
-  logLaraData(laraData) {
-    if (this.collaboratorUrls != null ? this.collaboratorUrls.length : undefined) { laraData.collaboratorUrls = this.collaboratorUrls }
+  logLaraData(laraData: CFMLaraProviderLogData) {
+    if (this.collaboratorUrls?.length) { laraData.collaboratorUrls = this.collaboratorUrls }
     if (this.options.logLaraData) { this.options.logLaraData(laraData) }
     return this.client.log('logLaraData', laraData)
   }
 
   // don't show in provider open/save dialogs
-  filterTabComponent(capability, defaultComponent) {
+  filterTabComponent(capability: ECapabilities, defaultComponent: React.Component): React.Component | null {
     return null
   }
 
-  extractRawDataFromRunState(runState) {
+  extractRawDataFromRunState(runState: any) {
```

Comment:
I think this would be better typed as `runState: string|object`

Reply:
The runState was one of those things that seemed to warrant a more thorough investigation so I left it alone.